File: backend/src/main/java/com/project/backend/domain/notification/repository/NotificationRepository.java
package com.project.backend.domain.notification.repository;

import com.project.backend.domain.notification.dto.NotificationDTO;
import com.project.backend.domain.notification.entity.Notification;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;


/**
 * 알림 레파지토리
 */
@Repository
public interface NotificationRepository extends JpaRepository<Notification,Long> {


    List<NotificationDTO> findAllByConsumerMemberId(Long id);
}

File: backend/src/main/java/com/project/backend/domain/notification/repository/NotificationRepository.kt
package com.project.backend.domain.notification.repository

import com.project.backend.domain.notification.entity.Notification
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository

/**
 * 알림 레포지토리
 */
interface NotificationRepository : JpaRepository<Notification, Long> {

    fun findAllByConsumerMemberId(id: Long): List<Notification>

    fun findAllByConsumerMemberId(id: Long, pageable: Pageable): Page<Notification>

    fun findAllByConsumerMemberIdAndIsCheckFalse(id: Long, pageable: Pageable): Page<Notification>

    fun countByConsumerMemberId(memberId: Long): Long
}
